xmtp_mls/src/groups/group_permissions.rs

```diff
@@ -231,19 +231,22 @@ pub struct PolicySet {
     pub remove_member_policy: MembershipPolicies,
     pub add_installation_policy: MembershipPolicies,
     pub remove_installation_policy: MembershipPolicies,
+    pub update_group_name_policy: MembershipPolicies,
```

Comment:
I kinda think we should have a new type of policy for metadata. Some of the `MembershipPolicies` aren't applicable here (for example, `AllowSameMember`) and it makes easier to be misused having that as an option.

What if we had a different type, like `MetadataPolicy` that can be more tailored towards updating metadata.

Reply:
Good point, ill make that update 👍

Reply:
@neekolas updating to include a new `MetadataPolicy` type actually adds a lot of cascading updates, since it touches all the validation logic. I was going to tackle validation logic in part 3, so how do we feel about leaving MetadataPolicy to the part 3 PR?